src/middleware/errorHandler.ts

```diff
@@ -52,7 +52,7 @@ const getNameForError = (error: unknown): string => {
 
 const getHttpStatusCodeForError = (error: unknown): number => {
   if (error instanceof DatabaseError) {
-    const errorCode = error.tinyPgError.queryContext.error.code;
+    const errorCode = error.tinyPgError.queryContext.error.code as PostgresErrorCode;
```

Comment:
I considered doing this cast on the `switch` statement itself inside of the helper functions, but decided that it was stronger to signal type at the parameter level and to perform the casting at the point of data access.

Reply:
Following up here @jasonaowen...

On reflection this does still feel a little gnarly.  Technically errorCode *could* hold a value that is not defined in PostgresErrorCode.  Maybe what I need to do is create a helper function along the lines of `getPostGresErrorCode` which takes a string and maps it to the proper enum value (and then add an `unknown`) case.

The thing that's frustrating is that the original switch statements work just fine, and naturally covered all possible values for `errorCode`.  I suppose I'm not sure the problem that the lint rule is trying to avoid.

Reply:
Decided to refactor this a bit!